ecosystem/sep-0012.md

```diff
@@ -503,6 +500,58 @@ Name | Type | Description
 | Client not authenticated properly         | `401 Unauthorized` |
 | Anchor has no information on the customer | `404 Not Found`    |
 
+### Callback POST Request
+
+```
+POST [url from PUT request]
+```
+
+See [`GET /customer reponse`](#response) for the POST request fields.
+
+
+In order to validate the integrity and provenance of the request, the Anchor MUST include a signature in an additional HTTP Header `X-Stellar-Signature`.
+This Header MUST follow the specification: `X-Stellar-Signature: t=<timestamp>, s=<base64 signature>` where:
+* __timestamp__ is the current timestamp at the time the callback is sent. This is used to assure the freshness of the request and to prevent this request to be replayed in the future. 
+* __base64 signature__ is the base64 encoding of the request signature. We explain below how to compute and verify this signature. The signature is computed using the Stellar private key linked to the `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md). Note that the timestamp and the HTTP Header Host will be part of the signature to prevent replay and relay attacks.
+
+It is the wallet's responsability to:
+* Verify the signature using the corresponding Stellar `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md).
+* Verify the freshness of the request by comparing the `timestamp` in the request with the current timestamp at the time of the reception and discard every request above a threshold of few seconds (1 or 2 minute(s) maximum).
+* Verify that the callback is not being relayed by validating that the HTTP Header __Host__ is equal to the Wallet Host.
+* Send a working callback URL to the anchor.
+
+### VERIFY signature
+
+* Check that callback request has `X-Stellar-Signature` header
+* Parse the header and extract:
+  * Key `t`: __timestamp__
+  * Key `s`: __base64 signature__
+* Verify the request freshness: _current timestamp_ - __timestamp__ < few seconds (1-2 minute(s) max)
+* Extract the HTTP Header __Host__, validate that it is equal to the Wallet Host
+* Extract the __body__ of the request
+* Base64 decode the __base64 signature__ to get the __signature__
+* Prepare the payload to verify the signature:
+  * The __timestamp__ (as a string)
+  * The character `.`
+  * The __Host__
+  * The character `.`
+  * The __body__
+* Verify the signature using the correct `SIGNING_KEY`
+
+### COMPUTE signature
+
+* Prepare the callback [request body](#response)
+* Prepare the payload to sign:
+  * Current timestamp (as a string)
+  * The character `.`
+  * The wallet Host to send the callback request to
+  * The character `.`
+  * The callback request body
+* Compute the signature using the Anchor private seed
```

Comment:
❓ which encryption algorithm should the anchor use to generate the signature with the private key?

Reply:
This is defined by the private seed. I think for stellar it can only be ed25519. This should be abstracted by the SDK.
For example in [Go](https://github.com/stellar/go/blob/b29a93b358b6669d951d4902c0bed6e1d67039fc/keypair/from_address.go#L56):
```go
import (
 b64 "encoding/base64"
 "github.com/stellar/go/keypair"
)

func sign(seed string, bufferToSign []byte) string {
  stellarKey := keypair.Parse(seed)
  signature := stellarKey.Sign(bufferToSign)
  return b64.StdEncoding.EncodeToString(signature)
}
```

Reply:
Ah I mistakenly read it as compute the hash.

Can we add the final payload here to make it more clear on how the concatenation should be done?
```suggestion
* Sign the payload '<timestamp>.<host>.<body>' using the Anchor private key
```